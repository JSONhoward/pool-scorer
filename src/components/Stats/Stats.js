import React, {useContext} from 'react'
import styled from 'styled-components'
import { StraightPoolContext } from '../../store'

const StatDiv = styled.div`
display: grid;
place-items: center;
width: 90%;
font-size: 1.5rem;
margin: 0 auto;
`

const Stats = () => {
    // eslint-disable-next-line
    const [straightPoolState, straightPoolDispatch] = useContext(StraightPoolContext)
    const {ballsRemaining, currentRun, fouls1, fouls2, runs1, runs2, player1, players} = straightPoolState

    return (
        <>
        <StatDiv><p>Balls Remaining: {ballsRemaining}</p></StatDiv>
        <StatDiv><p>Current Run: {currentRun}</p></StatDiv>
        {
            players === 1 ?
                (
                    <>
                        <StatDiv><p>High Run: {Math.max(...runs1)}</p></StatDiv>
                        <StatDiv><p>On Foul: {fouls1}</p></StatDiv>
                    </>
                )
                :
                (
                    <>
                        <StatDiv><p>High Run: {player1 ? Math.max(...runs1) : Math.max(...runs2)}</p></StatDiv>
                        <StatDiv><p>On Foul: {player1 ? fouls1 : fouls2}</p></StatDiv>
                    </>
                )
        }
        </>
    )
}

export default Stats
