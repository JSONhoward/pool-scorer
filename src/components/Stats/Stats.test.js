import React from 'react'
import {act, create} from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import {unmountComponentAtNode, render} from 'react-dom'
import Stats from './Stats'

let container = null
beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
})

afterEach(() => {
    unmountComponentAtNode(container)
    container.remove()
    container = null
})

describe('Stats Tests', () => {
    const props = {
        stat: 12,
        statLabel: 'test label'
    }
    test('Snapshot Test', () => {
        const tree = create(<Stats />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const component = create(<Stats {...props} />).root

        expect(component.props).toEqual(props)
    })

    test('Renders Props', () => {
        act(() => {
            render(<Stats {...props} />, container)
        })
        
        expect(container.textContent).toEqual(expect.stringContaining(props.statLabel))
    })
})