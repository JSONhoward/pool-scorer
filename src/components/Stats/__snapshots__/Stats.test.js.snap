// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Tests Snapshot Test 1`] = `
<p
  className="sc-AxjAm jjvHpt"
>
  : 
</p>
`;
